### PR TITLE
API: Update `asarray` function

### PR DESCRIPTION
**Pull request by mtsokol**

### CODE REVIEW

File: sparse/pydata_backend/_compressed/compressed.py [hunk 1]
```diff
@@ -857,14 +868,19 @@ class CSR(_Compressed2d):
     Sparse supports 2-D CSR.
     """
 
-    def __init__(self, arg, shape=None, prune=False, fill_value=0):
-        super().__init__(arg, shape=shape, compressed_axes=(0,), fill_value=fill_value)
+    def __init__(self, arg, shape=None, compressed_axes=(0,), prune=False, fill_value=0):
+        assert compressed_axes == (0,), "CSR only accepts rows as compressed axis"
```
Comment (hameerabbasi) [inline sparse/pydata_backend/_compressed/compressed.py:872]: Let's replace this by `ValueError`.

File: sparse/pydata_backend/_compressed/compressed.py [hunk 2]
```diff
@@ -882,14 +898,19 @@ class CSC(_Compressed2d):
     Sparse supports 2-D CSC.
     """
 
-    def __init__(self, arg, shape=None, prune=False, fill_value=0):
-        super().__init__(arg, shape=shape, compressed_axes=(1,), fill_value=fill_value)
+    def __init__(self, arg, shape=None, compressed_axes=(1,), prune=False, fill_value=0):
+        assert compressed_axes == (1,), "CSR only accepts columns as compressed axis"
```
Comment (hameerabbasi) [inline sparse/pydata_backend/_compressed/compressed.py:902]: Same here.

**Review by hameerabbasi — CHANGES_REQUESTED** [review]
Small changes.

File: sparse/pydata_backend/_compressed/compressed.py [hunk 3]
```diff
@@ -847,6 +848,16 @@ def __str__(self):
     def ndim(self) -> int:
         return 2
 
+    @classmethod
+    @abstractmethod
+    def get_compressed_axes(cls):
+        pass
```
Comment (hameerabbasi) [inline sparse/pydata_backend/_compressed/compressed.py:854]: Do we need this method (and the ones in the classes below)?

File: sparse/pydata_backend/_compressed/compressed.py (hunk 1 above)
Comment (mtsokol) [reply sparse/pydata_backend/_compressed/compressed.py]: Done!

File: sparse/pydata_backend/_compressed/compressed.py (hunk 2 above)
Comment (mtsokol) [reply sparse/pydata_backend/_compressed/compressed.py]: Done!

File: sparse/pydata_backend/_compressed/compressed.py (hunk 3 above)
Comment (mtsokol) [reply sparse/pydata_backend/_compressed/compressed.py:854]: I think yes. `from_coo` that is called in `from_numpy` needs to have `compressed_axes`. If it's `None` then it selects one based on best compression ratio. So there must be this piece of information in `CSC` and `CSR` classes, namely which `compressed_axes` should be used.

Comment (hameerabbasi) [reply sparse/pydata_backend/_compressed/compressed.py:854]: There's already the `compressed_axes` attribute/property for each instance.

Comment (mtsokol) [reply sparse/pydata_backend/_compressed/compressed.py:854]: But it's an instance attribute - available once object is created. Here we want to make sure that `compressed_axes` is exactly as the class require it (therefore we need a class attribute here for `CSC` and `CSR`, to make sure we have correct `compressed_axes` on creation).

Comment (mtsokol) [reply sparse/pydata_backend/_compressed/compressed.py:854]: Alternatively we could have copied `def from_coo` for `CSR`/`CSC` that just hardcode `compressed_axes` but I think it isn't clean approach. I would stick to class attribute that says what `compressed_axes` is.

Comment (hameerabbasi) [reply sparse/pydata_backend/_compressed/compressed.py:854]: In that case, let's use a class attribute rather than a class method.

Comment (mtsokol) [reply sparse/pydata_backend/_compressed/compressed.py]: Sure - done!

**Review by hameerabbasi — APPROVED** [review]

**Pull request merged**